### PR TITLE
Add pymagicc

### PR DESCRIPTION
**Pull request by znicholls**
Adds Pymagicc, a Python wrapper for running the widely used Model for the Assessment of Greenhouse Gas Induced Climate Change (MAGICC)

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by znicholls** [comment]
@rgieseke you know the drill :)

**Comment by rgieseke** [comment]
Happy to help maintain!

File: recipes/pymagicc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
```
Comment (npavlovikj) [inline recipes/pymagicc/meta.yaml]: I think the build error occurs for Python 3.8, so can you please try with `python >=3.5,<3.8` in both `host` and `run` sections?

Comment (znicholls) [reply recipes/pymagicc/meta.yaml]: I haven't filled in the sha256 bit yet (cause we need to do an sdist release on pypi), that's probably it right?

Comment (npavlovikj) [reply recipes/pymagicc/meta.yaml]: Ah, yes, you are correct. You can use the release tarball from GitHub it you want, to test your recipe until the PyPi tarball is uploaded.

File: recipes/pymagicc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - f90nml
+    - PyYAML
+    - scmdata >=0.2.1
+
+test:
+  imports:
+    - pymagicc
+
+  commands:
+    python -c "import pymagicc; assert pymagicc.__version__ == '2.0.0-beta.11'"
```
Comment (marcelotrevisani) [inline recipes/pymagicc/meta.yaml]:
```suggestion
    python -c "import pymagicc; assert pymagicc.__version__ == {{ version }}"
```

File: recipes/pymagicc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+about:
+  home: https://github.com/openclimatedata/pymagicc
+  license: AGPLv3
```
Comment (marcelotrevisani) [inline recipes/pymagicc/meta.yaml]:
```suggestion
  license: AGPL-3.0
```
https://opensource.org/licenses/AGPL-3.0

Comment (rgieseke) [reply recipes/pymagicc/meta.yaml]: Pymagicc wraps the CC-BY-NC-SA licensed MAGICC6 binary. Pymagicc itself is AGPL licensed.
Should this be noted somewhere/somehow in the metadata?

**Comment by znicholls** [comment]
@conda-forge/help-python could we request a review please?

The outstanding issue is summarised by @rgieseke 

> Pymagicc wraps the CC-BY-NC-SA licensed MAGICC6 binary. Pymagicc itself is AGPL licensed.
Should this be noted somewhere/somehow in the metadata?

**Comment by znicholls** [comment]
@conda-forge/help-python is there anything we can do to help with a review?

**Comment by znicholls** [comment]
@conda-forge/staged-recipes have we done something wrong which causes our review to be held up? (I don't mean to be pushy but just want to make sure we haven't missed anything as reviews have previously been so quick)

**Comment by chrisburr** [comment]
> Pymagicc wraps the CC-BY-NC-SA licensed MAGICC6 binary. Pymagicc itself is AGPL licensed. 
Should this be noted somewhere/somehow in the metadata?

Is the MAGICC6 binary being included in this package (1) or do you require that users figure out how to add it to their PATH (2)?

If (1) it might be that this package can't be `noarch: python` and you'll need to modify the metadata.  If (2) you don't need to do anything as this package isn't including anything for which `CC-BY-NC-SA` applies.

Regardless, have you considered adding a `magicc6` package? It looks like the licensing should be compatible with that (though maybe it's hard to get a download URL).

**Comment by znicholls** [comment]
@chrisburr thank you! Super helpful.

> Regardless, have you considered adding a `magicc6` package?

Unfortunately it was only ever released as a windows binary so users have to use wine (which isn't on conda I don't think). Our current solution is to package MAGICC6 in pymagicc but have a clear error if wine isn't installed and users try to run MAGICC (see https://github.com/openscm/pymagicc/blob/3c7f351f4cc286c9eb90ebf6dcaf62be81e6b65b/pymagicc/core.py#L338). 

So if wine isn't on conda, is this the best solution or would you recommend something else?

**Comment by chrisburr** [comment]
I thought I'd seen it all but that's an ingenious new way [to make package managers cry](https://www.youtube.com/watch?v=NSemlYagjIU). 😄

In that case your license expression should become `AGPL-3.0 AND CC-BY-NC-SA-3.0` and your should make sure the text of both licenses are included. If it's not already in a single file you can provide a list:
```yaml
   license_file:
       - LICENSE
       - LICENSE-MAGICC6
```

**Comment by znicholls** [comment]
> I thought I'd seen it all but that's an ingenious new way [to make package managers cry](https://www.youtube.com/watch?v=NSemlYagjIU)

Honestly it's a nightmare (fortunately our user base is sufficiently small that we haven't hurt too many package managers yet). We're planning on fixing it, but we need a new MAGICC version (MAGICC7) with OS specific binaries first...

Updated, thanks very much for the help.

**Comment by chrisburr** [comment]
> Honestly it's a nightmare (fortunately our user base is sufficiently small that we haven't hurt too many package managers yet). We're planning on fixing it, but we need a new MAGICC version (MAGICC7) with OS specific binaries first...

I can imagine! For the purposes of that video, you count as the "package manager" but it seems like you have a decent solution.

Looks good so LGTM! Thanks for the contribution.

**Comment by znicholls** [comment]
Awesome many thanks @chrisburr!!

**Pull request merged**